packages/adapter-utils/src/template.ts

```diff
@@ -25,9 +25,27 @@ export type TemplateContainer = {
     fieldName: string
   }
 
+export const compactString = (
+  part: TemplatePart, allParts: TemplatePart[]
+): void => {
+  const tempPart = allParts[allParts.length - 1]
+  if (_.isString(part) && _.isString(tempPart)) {
+    allParts.pop()
+    allParts.push(tempPart.concat(part))
+  } else {
+    allParts.push(part)
+  }
+}
+
+export const createTemplateExpression = (parts: {parts: TemplatePart[] }):TemplateExpression => {
+  const newParts:TemplatePart[] = []
+  parts.parts.forEach(part => compactString(part, newParts))
```

Comment:
it would be better if this could be
```suggestion
  const newParts = compactStrings(parts.parts)
```
especially if `compactStrings` is exported, I don't think we have a use case where we want to maintain `allParts` externally - @alonstern that'd be good for your use case as well, right?